src/logging.cpp

```diff
@@ -0,0 +1,77 @@
+#include "logging.hpp"
+
+#include "config.h"
+#include "util/files.hpp"
+
+#include <quill/Quill.h>
+
+#include <filesystem>
+#include <string>
+
+static const std::string LOGLINE_FORMAT =
+    "%(ascii_time) [%(thread)] [%(fileline:<18)] %(level_name) [%(logger_name:<14)] - "
+    "%(message)";
+
+namespace logging {
+
+quill::LogLevel _LOG_LEVEL;
+
+void
+init(quill::LogLevel log_level) {
+    _LOG_LEVEL = log_level;
+
+    // Create the logs directory
+    if (!std::filesystem::is_directory(log_dir())) {
+        std::filesystem::create_directory(log_dir());
+    }
+
+    // Create our config object
+    quill::Config cfg;
+
+    // Set main logger name
+    cfg.default_logger_name = "main";
+
+    // Initialize print handler
+    auto stdout_handler = dynamic_cast<quill::ConsoleHandler*>(quill::stdout_handler());
+    stdout_handler->set_pattern(
+        LOGLINE_FORMAT,
+        "%F %T.%Qms %z" // ISO 8601 but with space instead of T
+    );
+    stdout_handler->set_log_level(log_level);
+    stdout_handler->enable_console_colours();
+
+    cfg.default_handlers.emplace_back(stdout_handler);
+
+    // Initialize file handler
+#if DEBUG()
+    // Rotate through file handlers to save space
+    quill::Handler* file_handler = quill::rotating_file_handler(
+        log_dir() / "app.log", "a",
```

Comment:
noooo it was good the way it was. Everything in one file is unbelievably aggravating to read. I also don't think this should be a complied option. Should be in a user settings json (with things like controls).

Reply:
Devs will run the game in short bursts, so we want one big log file on dev.

Users will run a game in long chunks, so we want individual files per log.